### PR TITLE
SONAR-10168 Log downloading of plugins in debug

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginInstaller.java [hunk 1]
```diff
@@ -67,20 +67,58 @@ public ScannerPluginInstaller(ScannerWsClient wsClient, FileCache fileCache, Sca
...
+        }
+      }
+      System.out.println();
+    }
+
+    public void flagStop() {
+      stop = true;
+      try {
+        this.join();
+      } catch (InterruptedException e) {
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginInstaller.java]: MAJOR Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2142)

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginInstaller.java [hunk 2]
```diff
@@ -67,20 +67,58 @@ public ScannerPluginInstaller(ScannerWsClient wsClient, FileCache fileCache, Sca
...
+
+    private Progress() {
+      super("plugin-load-progress");
+    }
+
+    @Override
+    public void run() {
+      while (!stop) {
+        try {
+          System.out.print('.');
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginInstaller.java]: MAJOR Replace this use of System.out or System.err by a logger. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginInstaller.java [hunk 3]
```diff
@@ -67,20 +67,58 @@ public ScannerPluginInstaller(ScannerWsClient wsClient, FileCache fileCache, Sca
...
+          Thread.sleep(1000);
+        } catch (InterruptedException e) {
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginInstaller.java]: MAJOR Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2142)

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginInstaller.java [hunk 4]
```diff
@@ -67,20 +67,58 @@ public ScannerPluginInstaller(ScannerWsClient wsClient, FileCache fileCache, Sca
...
+          break;
+        }
+      }
+      System.out.println();
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerPluginInstaller.java]: MAJOR Replace this use of System.out or System.err by a logger. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 4 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by dbmeneses** [comment]
good point @henryju, i've dropped the usage of print

**Pull request merged**